src/parcsr_ls/par_relax_more.c

```diff
@@ -89,6 +89,12 @@ hypre_ParCSRMaxEigEstimate(hypre_ParCSRMatrix *A, /* matrix to relax with */
    hypre_MPI_Allreduce(&max_norm, &temp, 1, HYPRE_MPI_REAL, hypre_MPI_MAX, hypre_ParCSRMatrixComm(A));
    max_norm = temp;
 
+   hypre_MPI_Allreduce(&pos_diag, &temp, 1, HYPRE_MPI_INT, hypre_MPI_SUM, hypre_ParCSRMatrixComm(A));
```

Comment:
Hi @AerysL .  I don't think we should do this because it introduces a lot of extra communication and it shouldn't be a problem for the definite cases, which we more-or-less require in AMG.  If we are going to change something here, then we can keep track of the upper and lower bounds of the Gershgorin circles and allreduce them both in one call.  The current `temp` variable is the upper bound, if we add the negative of the lower bound, then we can use MPI_MAX to do both at the same time.  The sign of the diagonal is not needed anymore if we do this.

Note that I am not even sure how to use these bounds for Chebyshev smoothing if we have an indefinite system.

Reply:
Thanks @rfalgout . We redid it as per your suggestions. Please take another look. Thanks!